src/Microsoft.NetCore.Analyzers/Core/Security/UseAutoValidateAntiforgeryToken.cs

```diff
@@ -0,0 +1,282 @@
+                    !wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask, out var taskTypeSymbol) ||
+                    !wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.MicrosoftAspNetCoreMvcFiltersAuthorizationFilterContext, out var authorizationFilterContextTypeSymbol))
+                {
+                    return;
+                }
+
+                // A dictionary from method symbol to set of methods invoked by it directly.
+                // The bool value in the sub ConcurrentDictionary is not used, use ConcurrentDictionary rather than HashSet just for the concurrency security.
+                var callGraph = new ConcurrentDictionary<IMethodSymbol, ConcurrentDictionary<IMethodSymbol, bool>>();
+
+                // Ignore cases where a global anti forgery filter is in use.
+                var hasGlobalAntiForgeryFilter = false;
+
+                // Verify that validate anti forgery token attributes are used somewhere within this project,
+                // to avoid reporting false positives on projects that use an alternative approach to mitigate CSRF issues.
+                var usingValidateAntiForgeryAttribute = false;
+                var onAuthorizationAsyncMethodSymbols = new HashSet<IMethodSymbol>();
+                var actionMethodSymbols = new HashSet<(IMethodSymbol, string)>();
+
+                // Constructing callGraph.
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(
+                    (OperationBlockStartAnalysisContext operationBlockStartAnalysisContext) =>
+                    {
+                        var owningSymbol = operationBlockStartAnalysisContext.OwningSymbol;
+
+                        if (owningSymbol is IMethodSymbol methodSymbol)
+                        {
+                            var calledMethods = new ConcurrentDictionary<IMethodSymbol, bool>();
+                            callGraph.TryAdd(methodSymbol, calledMethods);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(operationContext =>
+                            {
+                                calledMethods.TryAdd((operationContext.Operation as IInvocationOperation).TargetMethod, true);
+                            }, OperationKind.Invocation);
+                        }
+                    });
+
+                // Holds if the project has a global anti forgery filter.
+                compilationStartAnalysisContext.RegisterOperationAction(operationAnalysisContext =>
+                {
+                    var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                    var methodSymbol = invocationOperation.TargetMethod;
+
+                    if (filterCollectionTypeSymbol.GetBaseTypesAndThis().Contains(methodSymbol.ContainingType) &&
```

Comment:
>filterCollectionTypeSymbol [](start = 24, length = 26)

I think you want to be looking through methodSymbol.ContainingType's base types, in case methodSymbol.ContainingType is derived from FilterCollection.